src/iuse_actor.cpp

```diff
@@ -361,8 +361,24 @@ void consume_drug_iuse::load( JsonObject &obj )
 
 void consume_drug_iuse::info( const item&, std::vector<iteminfo>& dump ) const
 {
+    std::string vits;
+    for( const auto &v : vitamins ) {
+        // only display vitamins that we actually require
+        if( g->u.vitamin_rate( v.first ) > 0 ) {
+            if( !vits.empty() ) {
+                vits += ", ";
+            }
+            int lo = int( v.second.first  / ( DAYS( 1 ) / float( g->u.vitamin_rate( v.first ) ) ) * 100 );
+            int hi = int( v.second.second / ( DAYS( 1 ) / float( g->u.vitamin_rate( v.first ) ) ) * 100 );
```

Comment:
> vitamin_rate  is a  const  method so it's possible (probable?)

That does not mean the function is pure (same input =\> same output). `player::vitamin_rate` could be implemented like this:
```C++
int player::vitamin_rate() const {
    return rand() % 2;
}
// Or
int player::vitamin_rate() const {
    g = new game();
    // next call to `g->u.vitamin_rate` happens on a new player object.
    ...
}
```
None of those likely, but the compiler can not guaranty that it won't happen. Well, except with full link time optimization, maybe.

I agree with codemime regarding the temporary variable and its name. `vitamin_rate` returns the vitamin rate, so it's fine to store that value in `vitamin_rate`. Actually, skip the `vitamin_` part. `rate` is perfectly fine in this context.